packages/backend/src/nest/storage/communityMetadata/communityMetadata.store.ts

```diff
@@ -0,0 +1,217 @@
+
+  constructor() {}
+
+  public async init(orbitDb: OrbitDB, localDbService: LocalDbService, emitter: EventEmitter) {
+    logger('Initializing community metadata key/value store')
+
+    this.orbitDb = orbitDb
+    this.localDbService = localDbService
+
+    // If the owner initializes the CommunityMetadataStore, then the
+    // ID would be undefined at this point when they first create the
+    // community. This would mean that they wouldn't be able to to
+    // validate an entry's ownership when writing or replicating.
+    //
+    // As a quick solution, we dynamically retrieve the owner's
+    // identity in the validation function. So after the owner sets it
+    // when calling updateCommunityMetadata, the validation function
+    // could then retrieve it. In our case, this is probably fine,
+    // because only the owner would be in the network by the time they
+    // first call updateCommunityMetadata.
+    //
+    // However, I think it might be simpler to pass the owner's
+    // OrbitDB identity when creating the store. For this we need to
+    // know at the time of initialization whether or not someone is
+    // the owner.
+    this.store = await this.orbitDb.keyvalue<CommunityMetadata>('community-metadata', {
+      replicate: false,
+      // Partially construct index so that we can include an
+      // IdentityProvider in the index validation logic.
+      // @ts-expect-error - OrbitDB's type declaration of OrbitDB lacks identity
+      Index: constructPartial(CommunityMetadataKeyValueIndex, [orbitDb.identity.provider, this.localDbService]),
+      accessController: {
+        write: ['*'],
+      },
+    })
+
+    this.store.events.on('write', (_address, entry) => {
+      logger('Saved community metadata locally')
+      emitter.emit(StorageEvents.COMMUNITY_METADATA_SAVED, entry.payload.value)
+    })
+
+    this.store.events.on('replicated', async () => {
+      logger('Replicated community metadata')
+      // @ts-expect-error - OrbitDB's type declaration of `load` lacks 'options'
+      // TODO: Is this necessary here?
+      await this.store.load({ fetchEntryTimeout: 15000 })
+      const meta = await this.getCommunityMetadata()
+      if (meta) {
+        emitter.emit(StorageEvents.COMMUNITY_METADATA_SAVED, meta)
+      }
+    })
+
+    // @ts-expect-error - OrbitDB's type declaration of `load` lacks 'options'
+    await this.store.load({ fetchEntryTimeout: 15000 })
+    logger('Loaded community metadata to memory')
+    const meta = await this.getCommunityMetadata()
+    if (meta) {
+      emitter.emit(StorageEvents.COMMUNITY_METADATA_SAVED, meta)
+    }
+  }
+
+  public getAddress() {
+    return this.store?.address
+  }
+
+  public async close() {
+    await this.store?.close()
+  }
+
+  public async updateCommunityMetadata(newMeta: CommunityMetadata): Promise<CommunityMetadata | undefined> {
+    try {
+      // TODO: Also check OrbitDB identity when updating community metadata
```

Comment:
It's still a todo. I think it's fine if it's not implemented. Perhaps it's not even helpful at all. I mainly focus on what I think is most important in the first draft but leave comments for stuff that could maybe improve it. In this case, if someone modified their version of quiet or there was a bug in our code, checking this could help us by throwing an error. But it shouldn't affect the functionality of the feature since only the owner should be updating the community metadata based on how we have written the rest of the code.